lib/di/di_module.dart

```diff
@@ -0,0 +1,13 @@
+import 'package:flutter_modular/flutter_modular.dart';
+
+import '../network/api_client.dart';
+import '../repository/comic_repository.dart';
+
+class DIModule extends Module {
+  @override
+  List<Bind> get binds => [
+        Bind.lazySingleton((i) => DioClient(), export: true),
+        Bind.lazySingleton((i) => ComicRepository(api: i.args.data),
```

Comment:
1. In order to use an interface it should look like Bind.lazySingleton<InterfaceName>(...).
2. You could replace i.args.data with i()